usermods/wireguard/wireguard.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <WireGuard-ESP32.h>
+
+#include "wled.h"
+
+class WireguardUsermod : public Usermod {
+   public:
+    void setup() { configTzTime(posix_tz.c_str(), ntpServerName); }
+
+    void connected() {
+        if (wg.is_initialized()) {
+            wg.end();
+        }
+    }
+
+    void loop() {
+        struct tm timeinfo;
+        if (!is_enabled || !getLocalTime(&timeinfo, 0)) {
+            return;
+        }
+
+        if (wg.is_initialized() || !WLED_CONNECTED) {
+            return;
+        }
+
+        if (preshared_key.length() < 5) {
+            wg.begin(local_ip, private_key.c_str(), endpoint_address.c_str(), public_key.c_str(), endpoint_port, NULL);
```

Comment:
I am not sure if this belongs to `loop()`. Perhaps a better place may be `connected()`?
`loop()` is executed at full possible rate which may be up to 5000-6000 times per second.